cmd/bosun/sched/template.go

```diff
@@ -200,15 +188,50 @@ func (c *Context) eval(v interface{}, filter bool, series bool, autods int) (exp
 		}
 	}
 	if series && e.Root.Return() != parse.TypeSeries {
-		return nil, "", fmt.Errorf("egraph: requires an expression that returns a series")
+		return nil, "", fmt.Errorf("need a series, got %T (%v)", e, e)
 	}
 	res, _, err := e.Execute(c.runHistory.Context, c.runHistory.GraphiteContext, c.schedule.Conf.LogstashElasticHost, nil, c.runHistory.Start, autods, c.Alert.UnjoinedOK, c.schedule.Search, c.schedule.Conf.AlertSquelched(c.Alert))
 	if err != nil {
-		return nil, "", fmt.Errorf("%s: %v", v, err)
+		return nil, "", fmt.Errorf("%s: %v", e, err)
 	}
 	return res.Results, e.String(), nil
 }
 
+func (c *Context) evalResult(r expr.ResultSlice, filter bool, series bool, autods int) (expr.ResultSlice, string, error) {
+	if filter {
```

Comment:
I thought they *had to be* different?
evalExpr filters by modifying the query and querying for a narrower set of results, because that's what we want for performance (that's why this behavior got added a few months ago, no?)
evalResult filters by filtering the resultslice, because it only has a resultslice, no expression
i don't see how to make it "the same" code.

series check i could unify i guess, but then we might execute the expression pointlessly in evalExpr, only to find out the result is no good, whereas we could have found this out by inspecting the AST as we do now.